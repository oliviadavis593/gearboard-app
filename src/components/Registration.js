import React from 'react';
import '../styles/Registration.css'

function Registration() {
    return(
        <div>
            <form>
            <h2>Registration</h2>
                <div>
                    <label>First Name</label>
                    <input 
                    type='text'
                    name='first_name'
                    placeholder='Johnny Doe'
                    required
                    />
                </div>
                <div>
                    <label>Email</label>
                    <input 
                    type='text'
                    name='email'
                    placeholder='gearboard123@gmail.com'
                    required
                    />
                </div>
                <div>
                    <label>Password</label>
                    <input 
                    type='text'
                    name='password'
                    placeholder='gearBoardgreat!'
                    required
                    />
                </div>
                <div>
                    <label>Confirm Password</label>
                    <input 
                    type='text'
                    name='re-password'
                    placeholder='gearBoardgreat!'
                    required
                    />
                </div>
                <div>
                    <button type='submit'>
                        Login
                    </button>
                    <button>
                        Demo
                    </button>
                </div>
            </form>
        </div>
    )
}

export default Registration; 