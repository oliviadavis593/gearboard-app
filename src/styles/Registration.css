
.gb-registration__form {
    border: 1px solid #fff; 
    width: 300px; 
    height: 500px; 
    margin: 30px auto;
    background-color: #eee; 
}

.gb-register-container__buttons {
    margin-left: 80px;  
}

.gb-register__button {
    margin-right: 20px; 
}