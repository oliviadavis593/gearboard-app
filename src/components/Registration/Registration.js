import React, { Component } from 'react';
import LandingHeader from '../../views/LandingHeader'
import AuthApiService from '../../services/auth-api-service'
import '../../styles/Registration.css'

export default class Registration extends Component  {
    static defaultProps = {
        onRegistrationSuccess: () => {}
    }

    state = { error: null }

    handleSubmit = ev => {
        ev.preventDefault()
        const { full_name, email, password } = ev.target

        this.setState({ error: null })
        AuthApiService.postUser({
            full_name: full_name.value, 
            email: email.value, 
            password: password.value, 
        })
        .then(user => {
            full_name.value = '' 
            email.value = ''
            password.value = ''
            this.props.onRegistrationSuccess()
            this.props.history.push('/login')
        })
        .catch(res => {
            this.setState({ error: res.error })
        })
    }

    render() {
        const { error } = this.state
        return(
                <div>
                <header>
                    <LandingHeader />
                </header>
                <main role='main'>
                    <form 
                    className='gb-registration__form' 
                    onSubmit={this.handleSubmit}
                    >
                        <fieldset>
                            <legend><h2>Registration</h2></legend>
                            <label htmlFor='full_name'>Full Name:</label>
                            <input 
                            id='full_name'
                            type='text'
                            name='full_name'
                            required
                            />
    
                            <label htmlFor='email'>Email:</label>
                            <input 
                            id='email'
                            type='email'
                            name='email'
                            required
                            />
    
                            <label htmlFor='password'>Password:</label>
                            <input 
                            id='password'
                            type='password'
                            name='password'
                            required
                            />
    
                        </fieldset>
                        <div role='alert'>
                            {error && <p className='red'>{error}</p>}
                        </div>
                        <div className='gb-register-container__buttons'>
                        <button 
                            className='gb-register__button'
                            type='submit'
                        >
                                Register
                            </button>
                        </div>
                    </form>
                </main>
            </div>
        )
    }
}