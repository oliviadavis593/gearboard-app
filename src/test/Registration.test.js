import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Registration from '../components/Registration/Registration'


describe('Footer component', () => {
    it(`renders without crashing`, () => {
        const div = document.createElement('div')
        ReactDOM.render(<Registration />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Registration />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})