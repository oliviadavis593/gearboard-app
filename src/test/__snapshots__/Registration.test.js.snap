// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders the UI as expected 1`] = `
<div>
  <header>
    <header
      className="gb-landing__header"
    >
      <a
        aria-current="page"
        className="nav__link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <h1>
          GearBoard
        </h1>
      </a>
    </header>
  </header>
  <main
    role="main"
  >
    <form
      className="gb-registration__form"
      onSubmit={[Function]}
    >
      <fieldset>
        <legend>
          <h2>
            Registration
          </h2>
        </legend>
        <label
          htmlFor="full_name"
        >
          Full Name:
        </label>
        <input
          id="full_name"
          name="full_name"
          required={true}
          type="text"
        />
        <label
          htmlFor="email"
        >
          Email:
        </label>
        <input
          id="email"
          name="email"
          required={true}
          type="email"
        />
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <input
          id="password"
          name="password"
          required={true}
          type="password"
        />
      </fieldset>
      <div
        role="alert"
      />
      <div
        className="gb-register-container__buttons"
      >
        <button
          className="gb-register__button"
          type="submit"
        >
          Register
        </button>
      </div>
    </form>
  </main>
</div>
`;
